### PR TITLE
Large re-amp on the torchaudio/docs

### PR DESCRIPTION
**Pull request by jamarshon**
-With new features added and legacy code gets removed, the docs (rst) also have to be updated
-The current docs looks very shady/broken (some code missing e.g transforms forward) or formatted poorly
Preview: https://jamarshon.github.io/audio/

### CODE REVIEW

**Comment by zhangguanheng66** [comment]
LGTM

File: torchaudio/__init__.py [hunk 1]
```diff
@@ -205,11 +224,11 @@ def sox_signalinfo_t():
...
-      - rate (float), sample rate as a float, practically will likely be an integer float
-      - channel (int), number of audio channels
-      - precision (int), bit precision
-      - length (int), length of audio in samples * channels, 0 for unspecified and -1 for unknown
-      - mult (float, optional), headroom multiplier for effects and None for no multiplier
+        - rate (float), sample rate as a float, practically will likely be an integer float
+        - channel (int), number of audio channels
+        - precision (int), bit precision
+        - length (int), length of audio in samples * channels, 0 for unspecified and -1 for unknown
+        - mult (float, optional), headroom multiplier for effects and None for no multiplier
```
Comment (vincentqb) [inline torchaudio/__init__.py]: little detail: None or ``None``? Most places use ``None`` :)

File: torchaudio/functional.py [hunk 2]
```diff
@@ -58,7 +58,7 @@ def istft(stft_matrix,          # type: Tensor
 
     The n_frames, hop_length, win_length are all the same which prevents the calculation of right padding.
     These additional values could be zeros or a reflection of the signal so providing ``length``
-    could be useful. If ``length`` is ``None`` then padding will be aggressively removed
+    could be useful. If ``length`` is None then padding will be aggressively removed
```
Comment (vincentqb) [inline torchaudio/functional.py]: Same about None

File: torchaudio/kaldi_io.py [hunk 3]
```diff
@@ -21,17 +21,18 @@
...
-            and converts it to (file name/descriptor) -> generator(string, Tensor).
-        convert_contiguous (bool): determines whether the array should be converted into a
-            contiguous layout.
+    Args:
+        file_or_fd (str/FileDescriptor): File name or file descriptor
+        fn (Callable[[...], Generator[str, numpy.ndarray]]): Function that has the signature (
+            file name/descriptor) -> Generator(str, numpy.ndarray) and converts it to (
+            file name/descriptor) -> Generator(str, torch.Tensor).
+        convert_contiguous (bool): Determines whether the array should be converted into a
+            contiguous layout. (Default: None)
```
Comment (vincentqb) [inline torchaudio/kaldi_io.py]: Same

File: torchaudio/kaldi_io.py [hunk 4]
```diff
@@ -63,16 +63,15 @@ def read_vec_int_ark(file_or_fd):
 
 
 def read_vec_flt_scp(file_or_fd):
-    r"""Create generator of (key,vector<float32/float64>) tuples, read according to kaldi scp.
+    r"""Create generator of (key,vector<float32/float64>) tuples, read according to Kaldi scp.
 
-    Arguments:
-        file_or_fd (string/File Descriptor): scp, gzipped scp, pipe or opened file descriptor.
+    Args:
+        file_or_fd (str/FileDescriptor): Scp, gzipped scp, pipe or opened file descriptor
```
Comment (vincentqb) [inline torchaudio/kaldi_io.py]: lower case scp, since it's a command?

File: torchaudio/kaldi_io.py [hunk 5]
```diff
@@ -82,31 +81,29 @@ def read_vec_flt_scp(file_or_fd):
 def read_vec_flt_ark(file_or_fd):
     r"""Create generator of (key,vector<float32/float64>) tuples, which reads from the ark file/stream.
 
-    Arguments:
-        file_or_fd (string/File Descriptor): ark, gzipped ark, pipe or opened file descriptor.
+    Args:
+        file_or_fd (str/FileDescriptor): Ark, gzipped ark, pipe or opened file descriptor
```
Comment (vincentqb) [inline torchaudio/kaldi_io.py]: ark lower case?

File: torchaudio/kaldi_io.py [hunk 6]
```diff
@@ -82,31 +81,29 @@ def read_vec_flt_scp(file_or_fd):
...
 
 
 def read_mat_scp(file_or_fd):
-    r"""Create generator of (key,matrix<float32/float64>) tuples, read according to kaldi scp.
+    r"""Create generator of (key,matrix<float32/float64>) tuples, read according to Kaldi scp.
 
-    Arguments:
-        file_or_fd (string/File Descriptor): scp, gzipped scp, pipe or opened file descriptor.
+    Args:
+        file_or_fd (str/FileDescriptor): Scp, gzipped scp, pipe or opened file descriptor
```
Comment (vincentqb) [inline torchaudio/kaldi_io.py]: scp lower case?

File: torchaudio/kaldi_io.py [hunk 7]
```diff
@@ -116,14 +113,13 @@ def read_mat_scp(file_or_fd):
 def read_mat_ark(file_or_fd):
     r"""Create generator of (key,matrix<float32/float64>) tuples, which reads from the ark file/stream.
 
-    Arguments:
-        file_or_fd (string/File Descriptor): ark, gzipped ark, pipe or opened file descriptor.
+    Args:
+        file_or_fd (str/FileDescriptor): Ark, gzipped ark, pipe or opened file descriptor
```
Comment (vincentqb) [inline torchaudio/kaldi_io.py]: Same, ark

File: torchaudio/kaldi_io.py (hunk 4 above)
Comment (jamarshon) [reply torchaudio/kaldi_io.py]: Hmm I'm not sure about the correct grammar here. scp and ark are both file extensions and so at the beginning of sentence would you put "Mp3 ...." or "mp3 ..."?

https://kaldi-asr.org/doc/io.html

File: torchaudio/__init__.py (hunk 1 above)
Comment (jamarshon) [reply torchaudio/__init__.py]: Done.

File: torchaudio/functional.py (hunk 2 above)
Comment (jamarshon) [reply torchaudio/functional.py]: Done.

File: torchaudio/kaldi_io.py (hunk 3 above)
Comment (jamarshon) [reply torchaudio/kaldi_io.py]: Done.

**Comment by vincentqb** [comment]
MelScale renders a little strange here: https://jamarshon.github.io/audio/transforms.html#melscale

**Comment by vincentqb** [comment]
MelSpectrogram has inconsistent ":" in subtitles.
https://jamarshon.github.io/audio/transforms.html#melspectrogram

**Comment by vincentqb** [comment]
MuLaw Decoding has inconsistent indentations in subtitles.
https://jamarshon.github.io/audio/transforms.html#mulawdecoding

**Comment by vincentqb** [comment]
When you give the shape of tensors, you should be as explicit as possible for both input and output of `forward`. I'd go for 
"a tensor of shape `(Channels, frequency, time)`". I'm leaning for not using abbreviations.

File: torchaudio/transforms.py [hunk 8]
```diff
@@ -158,26 +158,26 @@ class MelSpectrogram(torch.jit.ScriptModule):
...
+        win_length (int): Window size. (Default: ``n_fft``)
         hop_length (int, optional): Length of hop between STFT windows. (
-            Default: `win_length // 2`)
-        n_fft (int, optional): Size of fft, creates `n_fft // 2 + 1` bins
-        f_min (float): Minimum frequency. (Default: 0.)
-        f_max (float, optional): Maximum frequency. (Default: `None`)
-        pad (int): Two sided padding of signal. (Default: 0)
-        n_mels (int): Number of mel filterbanks. (Default: 128)
+            Default: ``win_length // 2``)
+        n_fft (int, optional): Size of fft, creates ``n_fft // 2 + 1`` bins
```
Comment (vincentqb) [inline torchaudio/transforms.py]: Let's use FFT when referring to the algorithm.

**Comment by vincentqb** [comment]
Do we settle "(channel, ...)" or "(channels, ...)" ? :) I'm leaning for singular, since all other words are singular. Thoughts?

**Comment by vincentqb** [comment]
Similarly, do we settle on "a tensor of size (...)" or "a tensor of shape (...)"? That's one of those size/shape/dimension discussions :)

**Comment by vincentqb** [comment]
> Do we settle "(channel, ...)" or "(channels, ...)" ? :) I'm leaning for singular, since all other words are singular. Thoughts?

Also, I noticed we had agreed on `n_freqs` to refer to frequency, so we should update that in the documentation. But then this is a little inconsistent with `time` and `channel`. Should we have `n_times` and `n_channels` or leave it? Thoughts?

**Comment by vincentqb** [comment]
Let's use the convention described in #169 for the comments quoted below now that we've settled on it.

> Do we settle "(channel, ...)" or "(channels, ...)" ? :) I'm leaning for singular, since all other words are singular. Thoughts?

channel, and singular words

> Similarly, do we settle on "a tensor of size (...)" or "a tensor of shape (...)"? That's one of those size/shape/dimension discussions :)

dimensions (for "dimension names") and size for sizes (we'll avoid shape as much as we can)

> Also, I noticed we had agreed on `n_freqs` to refer to frequency, so we should update that in the documentation. But then this is a little inconsistent with `time` and `channel`. Should we have `n_times` and `n_channels` or leave it? Thoughts?

n_freq, n_channel, n_time, etc. for sizes

**Pull request merged**